src/modalities/nn/weight_init/low_level_weight_init_factory.py

```diff
@@ -0,0 +1,121 @@
+import math
+from typing import Annotated, List, Optional
+
+from pydantic import BaseModel, Field, model_validator
+
+from modalities.config.pydanctic_if_types import PydanticWeightInitializationIFType
+from modalities.nn.weight_init.weight_init import (
+    ModulewiseNormalInitialization,
+    NamedParameterwiseNormalInitialization,
+    WeightInitializationIF,
+    WeightInitializerWrapper,
+)
+
+
+class WeightInitializerWrapperConfig(BaseModel):
+    weight_initializers: List[PydanticWeightInitializationIFType]
+
+
+class PlainWeightInitializationConfig(BaseModel):
+    mean: float
+    std: Annotated[float, Field(strict=True, ge=0.0)] | str  # can be float or "auto"
+    hidden_dim: Optional[int] = None
+
+    @model_validator(mode="after")
+    def check_std_and_hidden_dim(self):
+        if self.std == "auto" and self.hidden_dim is None:
+            raise ValueError("hidden_dim must be specified when std is 'auto'")
+        return self
+
+
+class NamedParameterwiseNormalInitializationConfig(BaseModel):
+    mean: float
+    std: Annotated[float, Field(strict=True, ge=0.0)] | str  # can be float or "auto"
+    parameter_name_suffixes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class ScaledWeightInitializationConfig(BaseModel):
+    mean: float
+    plain_std: Annotated[float, Field(strict=True, ge=0.0)]
+    num_layers: Annotated[int, Field(strict=True, gt=0)]
+    parameter_name_suffixes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class ScaledEmbedInitializationConfig(BaseModel):
+    mean: float
+    parameter_name_suffixes: List[str]  # here we filter for the parameter names, e.g., "c_proj.weight"
+
+
+class LowLevelInitializationFactory:
+    @staticmethod
+    def get_weight_initializer_wrapper(weight_initializers: List[WeightInitializationIF]) -> WeightInitializationIF:
+        initializer_wrapper = WeightInitializerWrapper(weight_initializers)
+        return initializer_wrapper
+
+    @staticmethod
+    def get_plain_initialization(
+        mean: float, std: float | str, hidden_dim: Optional[int] = None
+    ) -> WeightInitializationIF:
+        """Initializes the weights of a model by sampling from a normal distribution.
+        NOTE: This class supports the initialization of nn.Linear and nn.Embedding layers.
+        For other layer types, the initialization must be subclassed and extended
+
+        Args:
+            mean (float): mean of the normal distribution
+            std (float): standard deviation of the normal distribution
+            hidden_dim (Optional[int], optional): hidden dimension of the attention layer. Defaults to None.
+        """
+
+        # auto: choose std automatically
+        if std == "auto":
+            if hidden_dim is None:
```

Comment:
`std` can be `auto` also for initialization types other than `plain`. So if we want to check this here (we seem to check it in 3 different places in total: `ComposedWeightInitializationConfig`, `PlainWeightInitializationConfig `, `LowLevelInitializationFactory`), we need to do it for `scaled` and `scaled_embed` as well.

Reply:
we now pass the std calculated from `plain` (i.e., working_std) to `scaled`.
See here: https://github.com/Modalities/modalities/blob/824da6ad8fed1fcf0e328d7bffc53ed4c3ffc072/src/modalities/nn/weight_init/high_level_weight_init_factory.py#L106

Therfore only `PlainInitializationConfig` has to allow the "auto" option and the other configs (i.e., `ScaledInitializationConfig` and `ScaledEmbedInitializationConfig`) accept only `float` for `std`

Reply:
Ok, I guess that works as well, as long as the `weight_initialization` component with `variant_key = composed` is always used in the config, not the other `variant_key`s, right?

Reply:
Yes, for the other variant key `scaled`, we would have to specify `std` in the config as float. I think this is not a limitation, because anyone can still use the `composed` one.

Reply:
That's fine, I just wonder whether we should allow people to use the variant keys `plain`, `scaled`, `scaled_embed` (and `wrapper`?) in a config file at all. You can specify everything you want using the variant key `composed`, so why not make it the only one "available" for the sake of simplicity?